codegenerator/cli/src/hbs_templating/codegen_templates.rs

```diff
@@ -367,15 +368,41 @@ impl EventMod {
             )),
         };
 
+        let (block_type, block_schema, transaction_type, transaction_schema) =
+            match self.custom_field_selection {
+                Some(ref field_selection) => {
+                    let field_selection = FieldSelection::new(FieldSelectionOptions {
+                        transaction_fields: field_selection.transaction_fields.clone(),
+                        block_fields: field_selection.block_fields.clone(),
+                        transaction_type_name: "transaction".to_string(),
+                        block_type_name: "block".to_string(),
+                    });
+                    (
+                        field_selection.block_type,
+                        field_selection.block_schema,
+                        field_selection.transaction_type,
+                        field_selection.transaction_schema,
+                    )
+                }
+                None => (
+                    "Block.t".to_string(),
```

Comment:
Am I understanding right that this Block.t refers to the global block field selection? So if users don't use specialised selections they can still rely on shared types for composing functions etc?

Reply:
Exactly. This is especially important for ReScript typesystem